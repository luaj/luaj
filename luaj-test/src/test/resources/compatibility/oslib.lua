-- simple os-library tests
--
-- because the nature of the "os" library is to provide os-specific behavior,
-- the compatibility tests must be extremely loose, and can really only 
-- compare things like return value type to be meaningful.
-- 
-- actual os behavior needs to go in an oslib function test
-- 
local pcall = function(...)
	local s,e,f = pcall(...)
	return s,type(e),type(f)
end 
print( 'os', type(os) )
print( 'os.clock()', pcall( os.clock ) )
print( 'os.date()', pcall( os.date ) )
print( 'os.difftime(123000, 21500)', pcall( os.difftime, 123000, 21250 ) )
print( 'os.getenv()', pcall( os.getenv ) )
print( 'os.getenv("bogus.key")', pcall( os.getenv, 'bogus.key' ) )
local s,p = pcall( os.tmpname )
local s,q = pcall( os.tmpname )
print( 'os.tmpname()', s, p )
print( 'os.tmpname()', s, q )
-- permission denied on windows
--print( 'os.remove(p)', pcall( os.remove, p ) )
--print( 'os.rename(p,q)', pcall( os.rename, p, q ) )
local s,f = pcall( io.open, p,"w" )
print( 'io.open', s, f )
print( 'write', pcall( f.write, f, "abcdef 12345" ) )
print( 'close', pcall( f.close, f ) )
print( 'os.rename(p,q)', pcall( os.rename, p, q ) )
print( 'os.remove(q)', pcall( os.remove, q ) )
print( 'os.remove(q)', pcall( os.remove, q ) )
-- setlocale not supported on jse yet
-- print( 'os.setlocale()', pcall( os.setlocale ) )
-- print( 'os.setlocale("jp")', pcall( os.setlocale, "jp" ) )
-- print( 'os.setlocale("us","monetary")', pcall( os.setlocale, "us", "monetary" ) )
-- print( 'os.setlocale(nil,"all")', pcall( os.setlocale, nil, "all" ) )
print( 'os.setlocale("C")', pcall( os.setlocale, "C" ) )
print( 'os.exit', type(os.exit) )

-- os.date() formatting
local t = 1281364496  -- Aug 9, 2010, 2:34:56 PM (Monday)
local function p(s)
  if pcall(os.date, s, t) then
	  print( "os.date('"..s.."', "..t..")", pcall(os.date, s, t))
  end
end
for i= 65, 90 do
  p('%'..string.char(i + 97 - 65))
  p('%'..string.char(i))
end
local tbl = os.date('*t', t)
for i,k in ipairs({'year', 'month', 'day', 'hour', 'min', 'sec', 'wday', 'yday', 'isdst'}) do
	local v = tbl[k]
	print('k', type(k), k, 'v', type(v), v)
end

print('type(os.time())', type(os.time()))
print('os.time({year=1971, month=2, day=25})', os.time({year=1971, month=2, day=25}))
print('os.time({year=1971, month=2, day=25, hour=11, min=22, sec=33})', os.time({year=1971, month=2, day=25, hour=11, min=22, sec=33}))
