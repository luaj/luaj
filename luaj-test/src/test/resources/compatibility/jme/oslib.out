os	table
os.clock()	true	number	nil
os.date()	true	string	nil
os.difftime(123000, 21500)	true	number	nil
os.getenv()	false	string	nil
os.getenv("bogus.key")	true	nil	nil
os.tmpname()	true	string
os.tmpname()	true	string
io.open	true	userdata
write	false	string	nil
close	false	string	nil
os.rename(p,q)	true	boolean	nil
os.remove(q)	true	boolean	nil
os.remove(q)	true	nil	string
os.setlocale("C")	true	string	nil
os.exit	function
os.date('%a', 1281364496)	true	string	nil
os.date('%A', 1281364496)	true	string	nil
os.date('%b', 1281364496)	true	string	nil
os.date('%B', 1281364496)	true	string	nil
os.date('%c', 1281364496)	true	string	nil
os.date('%C', 1281364496)	true	string	nil
os.date('%d', 1281364496)	true	string	nil
os.date('%D', 1281364496)	true	string	nil
os.date('%e', 1281364496)	true	string	nil
os.date('%F', 1281364496)	true	string	nil
os.date('%g', 1281364496)	true	string	nil
os.date('%G', 1281364496)	true	string	nil
os.date('%h', 1281364496)	true	string	nil
os.date('%H', 1281364496)	true	string	nil
os.date('%I', 1281364496)	true	string	nil
os.date('%j', 1281364496)	true	string	nil
os.date('%m', 1281364496)	true	string	nil
os.date('%M', 1281364496)	true	string	nil
os.date('%n', 1281364496)	true	string	nil
os.date('%p', 1281364496)	true	string	nil
os.date('%r', 1281364496)	true	string	nil
os.date('%R', 1281364496)	true	string	nil
os.date('%S', 1281364496)	true	string	nil
os.date('%t', 1281364496)	true	string	nil
os.date('%T', 1281364496)	true	string	nil
os.date('%u', 1281364496)	true	string	nil
os.date('%U', 1281364496)	true	string	nil
os.date('%V', 1281364496)	true	string	nil
os.date('%w', 1281364496)	true	string	nil
os.date('%W', 1281364496)	true	string	nil
os.date('%x', 1281364496)	true	string	nil
os.date('%X', 1281364496)	true	string	nil
os.date('%y', 1281364496)	true	string	nil
os.date('%Y', 1281364496)	true	string	nil
os.date('%z', 1281364496)	true	string	nil
os.date('%Z', 1281364496)	true	string	nil
k	string	year	v	number	2010
k	string	month	v	number	8
k	string	day	v	number	9
k	string	hour	v	number	16
k	string	min	v	number	34
k	string	sec	v	number	56
k	string	wday	v	number	2
k	string	yday	v	number	221
k	string	isdst	v	boolean	true
type(os.time())	number
os.time({year=1971, month=2, day=25})	36327600
os.time({year=1971, month=2, day=25, hour=11, min=22, sec=33})	36325353
